Feature/method tests

rlmoser99 opened the pull request:
Lesson: https://www.theodinproject.com/courses/ruby-programming/lessons/methods

Since students will be creating the entire method, I had to break away from the format of the previous exercises. I decided to incorporate methods that are helpful to know for upcoming exercises & projects.

In addition to describing what the method should do, I also included hints of the built-in method names. Although I do not want to spoon feed students, I do not want students to write a 10-line method when there is a built-in one. 

I was tempted to include a link to the built-in method documentation, but decided to just encourage students to look it up themselves. **Should I update the files with documentation links?**

`#ascii_translator`
`#alphabetical_list` - chaining methods
`#sports_in_common`
`#lucky_number` - normal and default parameters
`#ascii_code` - explicit and implicit returns
`#pet_pun` - using puts & if/else statement
`#twenty_first_century?` - predicate method

I did not include a bang method, because it does not look any different as the return value of a method. This would make more sense later in the curriculum using an instance variable.

The `#pet_pun` does not have a return value because it uses `puts`, therefore I included a test for the return value being nil. I think this test is important to show students. Since this test will output to the terminal, I stubbed it `allow_any_instance_of(Object).to receive(:puts).with("Cats are purr-fect!")` **Do you think this stub will cause confusion? Should we just let it output to the terminal?**

Closes #12

KevinMulhern commented:
These look really good @rlmoser99 💯 

> I was tempted to include a link to the built-in method documentation, but decided to just encourage students to look it up themselves. Should I update the files with documentation links?

I think encouraging the student to look up the docs themselves is the right move here 👍 

> Do you think this stub will cause confusion? Should we just let it output to the terminal?

I can see this maybe causing confusion if the student doesn't use puts. We could maybe move this into the before block if our main goal is to suppress the output in the console. 

```
  before(:each) do
    allow($stdout).to receive(:puts).with("Cats are purr-fect!")
    allow($stdout).to receive(:write)..with("Cats are purr-fect!")
  end
end
```

But if the goal is to test that they uses puts this is spot on the way it is already 👍

rlmoser99 commented:
Thanks @KevinMulhern for your feedback. Since this exercise is intending students to use `puts`, I made two small changes.
- added a hint to use puts (to eliminate the chance of confusion)
- changed the test to use `allow($stdout).to receive(:puts).with("Cats are purr-fect!")` because I think this will make more sense to the students. 

In addition, I rebased because of a merge conflict in the README. During the rebase, I noticed that I used the wrong lesson number for the hash exercises, so I fixed it.

KevinMulhern commented:
These exercises are great @rlmoser99, they will make the methods lesson so much better 💯  thanks for all your work on these.
